packages/apps/shopify-app-remix/src/server/authenticate/admin/scope/client/fetch-scopes-information.ts

```diff
@@ -0,0 +1,40 @@
+import {AdminApiContext} from '../../../../clients';
+
+export interface FetchScopeInformationResponse {
+  app: {
+    requestedAccessScopes: {
+      handle: string;
+    }[];
+    optionalAccessScopes: {
+      handle: string;
+    }[];
+    installation: {
+      accessScopes: {
+        handle: string;
+      }[];
+    };
+  };
+}
+
+const FETCH_SCOPE_INFORMATION_QUERY = `#graphql
+query FetchAccessScopes{
+  app {
+    requestedAccessScopes {
+      handle
+    }
+    installation {
+      accessScopes {
+        handle
+      }
+    }
+  }
+}`;
+
+export async function fetchScopeInformation(admin: AdminApiContext) {
+  const fetchScopeInformationResult = await admin.graphql(
+    FETCH_SCOPE_INFORMATION_QUERY,
+  );
+
+  return (await fetchScopeInformationResult.json())
+    .data as FetchScopeInformationResponse;
```

Comment:
Since `data` is typed `any`, it might be ever-so-slightly cleaner to make `Promise<FetchScopeInformationResponse>` the return type of the function instead. WDYT?